src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/docker/ConnectedContainer.kt

```diff
@@ -0,0 +1,30 @@
+package com.atlassian.performance.tools.sshubuntu.api.docker
+
+import com.github.dockerjava.api.DockerClient
+import com.github.dockerjava.api.command.ConnectToNetworkCmd
+
+class ConnectedContainer(
+    private val docker: DockerClient,
+    val containerId: String,
+    private val networkId: String
+) : AutoCloseable {
+
+    override fun close() {
+        docker
+            .disconnectFromNetworkCmd()
+            .withContainerId(containerId)
+            .withNetworkId(networkId)
+            .exec()
+    }
+}
+
+fun ConnectToNetworkCmd.execAsResource(
```

Comment:
Does it work from java?

Reply:
Yes, but it's not as syntactically sweet as in Kotlin:
```java
ConnectedContainerKt.execAsResource(cmd, docker)
```